### PR TITLE
[cherry-pick]Fix ParseSegmentIDBinlog panicks with bad input (#18413)

### PR DESCRIPTION
**Pull request by congqixia**
Cherry-pick from master
Related to #18412 

/kind bug

Signed-off-by: Congqi Xia <congqi.xia@zilliz.com>

### CODE REVIEW

File: internal/storage/binlog_util.go [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
 	"strings"
 )
 
+// ParseSegmentIDByBinlog parse segment id from binlog paths
+// if path format is not expected, returns error
 func ParseSegmentIDByBinlog(path string) (UniqueID, error) {
-	// binlog path should consist of "files/insertLog/collID/partID/segID/fieldID/fileName"
+	// binlog path should consist of "[prefix]/insertLog/collID/partID/segID/fieldID/fileName"
 	keyStr := strings.Split(path, "/")
+	if len(keyStr) != 7 {
```
Comment (xiaocai2333) [inline internal/storage/binlog_util.go:14]: Maybe changing to >= 7 is better? Because we can't decide what the user defines the `prefix` as.

Comment (jaime0815) [reply internal/storage/binlog_util.go:14]: Yes， we don't limit the prefixes to use x/x/x or no x/  within etcd configurations.  it should not support  x/x/... as a prefix if the length is 7

Comment (congqixia) [reply internal/storage/binlog_util.go:14]: currently binlog path format is fixed. When collector scans the oss directory, if the path does not matches the format we define, we shall return an error to ignore this file.

Comment (congqixia) [reply internal/storage/binlog_util.go:14]: Think twice, if prefix contains "/", the segment fetch logic cannot work

Comment (jaime0815) [reply internal/storage/binlog_util.go:14]: more prechecking of configuration should be implemented during the parameters initialize

**Pull request merged**